src/ui/Analytics/Analytics.ts

```diff
@@ -104,8 +104,13 @@ export class Analytics extends Component {
      * Default value is `https://platform.cloud.coveo.com/rest/ua`.
      */
     endpoint: ComponentOptions.buildStringOption({
-      defaultFunction: () => {
-        return AnalyticsEndpoint.getURLFromSearchEndpoint(SearchEndpoint.defaultEndpoint);
+      postProcessing: value => {
+        if (!value) {
+          return AnalyticsEndpoint.getURLFromSearchEndpoint(SearchEndpoint.defaultEndpoint);
+        }
+
+        const [basePlatform] = value.split('/rest');
+        return basePlatform + '/rest';
```

Comment:
Instead of splitting and re-appending, we could use a regex:
```ts
const match = /^.*?\/rest/.exec(value);
return match ? match[0] : value;
```

Reply:
We could use a regex, but I believe it would be more complex.

Note that when splitting a url not containing `/rest`, `split` still returns a value to which we append. This case is not handled by the above logic atm, and returns the url intact.